keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases.md

```diff
@@ -0,0 +1,152 @@
+monitor the networking performance metrics such as latency, throughput etc has arisen.
+
+We have come across 2 test frameworks which have limited testcases for network performance benchmarking.
+
+a) E2E (test\e2e\network\networking_perf.go): This framework has only 2 testcases for measuring throughput.
+
+b) netperf (https://github.com/kubernetes/perf-tests/tree/master/network/benchmarks/netperf) : This framework 
+   measures only the throughput between client-server pods for various MSS(Maximum Segment size). 
+   Other performance metrics such as latency , loss-rate , response time are not measured.
+
+### Goals
+
+The primary goal of this proposal is to enhance the testcases to cover various 
+networking scenarios :
+
+a)  IPv4 , IPv6 , Dual-Stack.
+b)  Protocols: TCP, UDP, HTTP/HTTPS.
+c)  With services such as ClusterIP , NodePort etc.
+
+We intend to submit tests to measure the following network performance metrics for various 
+scenarios such as different services (ClusterIP/NodePort), with IPv4/IPv6 configured etc.
+
+a)  Throughput (For TCP traffic)
+b)  Loss rate (For UDP traffic)
+c)  Latency/Jitter (For UDP/HTTP/HTTPS)
+d)  Response time (For HTTP/HTTPs)
+
+Our important goal is to make sure these benchmarking testcases are run periodically,
+so that any degradation in performance is detected in early phase of feature delivery.
+
+### Non-goals
+
+Benchmarking and measurements with intermediate services like ingress controllers, load-balancers 
+are out of scope of this proposal.
+
+### Challenges and Open Questions [optional]
+
+1) Which is the suitable hosting framework for proposed tests ? 
+2) Periodic execution of these proposed tests ?
+
+
+## Proposal
+
+### Test Scenarios
+
+As captured in the link https://docs.google.com/document/d/1_760ptAHBY_c56slWhjl6PJQOrf__vQFJXkL58ewg1M/edit
+we propose to have 2 broad categories based on how client PODs can communicate with server PODs.
+
+The testcase will cover different protocols such as TCP, UDP, HTTP/HTTPS with IPv4/IPv6 addresses configured.
+
+The testcases will cover different client-to-server pod number ratio (i.e 1:N , 1:1 , N:1)
+
+### Tools
+
+We intend to use the following tools for testing in addition to iPerf/netperf already being used
+
+a)  T-Rex: T-Rex is opensource traffic generator tool made available by CISCO.
+    It will be used for measuring the UDP maximum throughput (Kpps), RTT latency (min,max,avg).
+
+b)  Siege: Siege is an open source benchmark utility. 
+    It will be used for measuring HTTP/HTTPs metrics such as total number of hits recorded, 
+    bytes transferred, response time, concurrency, and return status. 
+
+	
+### Execution Environment
+
+We propose to enhance the testcases in either of the 2 test frameworks i.e E2E or netperf.
+
+Our recommendation is to include these testcases in kubernetes CI framework,so that these
+testcases are run periodically (once in a week). 
```

Comment:
Depending on the cost of those tests, we can run them more often. To estimate the cost (for e2e tests) we'll need to know the size of the cluster the tests will be running in and the time that the tests will take.

Reply:
Yes once we finalize the test framework, we can implement the various testcases and see the runtime. After that, we can decide whether to include all testcases or selective ones.